pkg/archive/archive.go

```diff
@@ -460,6 +460,20 @@ func createTarFile(path, extractDir string, hdr *tar.Header, reader io.Reader, L
 			return err
 		}
 	}
+
+	if hdr.Typeflag == tar.TypeReg || hdr.Typeflag == tar.TypeRegA {
+		file, err := os.Open(path)
+		if err != nil {
+			file.Close()
```

Comment:
If err is not nil then the file was not opened, so does not need to be closed in the error case. It also looks safe to do the close with a "defer file.Close()" once we know the file was successfully opened.